lib/job_queue/job_node.c

```diff
@@ -415,69 +415,89 @@ void job_queue_node_run_EXIT_callback( job_queue_node_type * node ) {
 
-      if (new_status == JOB_QUEUE_FAILED)
-        job_queue_node_fscanf_EXIT( node );
-    }
-  }
+  if (!(new_status & JOB_QUEUE_COMPLETE_STATUS))
+    return;
+
+  node->sim_end = time( NULL );
+  node->progress_timestamp = node->sim_end;
+
+  if (new_status == JOB_QUEUE_FAILED)
+    job_queue_node_fscanf_EXIT( node );
+
 }
 
 
-submit_status_type job_queue_node_submit( job_queue_node_type * node , job_queue_status_type * status , queue_driver_type * driver) {
+submit_status_type job_queue_node_submit(job_queue_node_type * node,
+                                         job_queue_status_type * status,
+                                         queue_driver_type * driver) {
   submit_status_type submit_status;
   pthread_mutex_lock( &node->data_mutex );
-  {
-    void * job_data = queue_driver_submit_job( driver,
-                                               node->run_cmd,
-                                               node->num_cpu,
-                                               node->run_path,
-                                               node->job_name,
-                                               node->argc,
-                                               (const char **) node->argv);
-    if (job_data != NULL) {
-      job_status_type old_status = node->job_status;
-      job_status_type new_status = JOB_QUEUE_SUBMITTED;
-
-      node->job_data = job_data;
-      node->submit_attempt++;
-      /*
-        The status JOB_QUEUE_SUBMITTED is internal, and not
-        exported anywhere. The job_queue_update_status() will
-        update this to PENDING or RUNNING at the next call. The
-        important difference between SUBMITTED and WAITING is
-        that SUBMITTED have job_data != NULL and the
-        job_queue_node free function must be called on it.
-      */
-      submit_status = SUBMIT_OK;
-      job_queue_node_set_status( node , new_status);
-      job_queue_status_transition(status, old_status, new_status);
-    } else
-      /*
-        In this case the status of the job itself will be
-        unmodified; i.e. it will still be WAITING, and a new attempt
-        to submit it will be performed in the next round.
-      */
-      submit_status = SUBMIT_DRIVER_FAIL;
+
+  void * job_data = queue_driver_submit_job( driver,
+                                             node->run_cmd,
+                                             node->num_cpu,
+                                             node->run_path,
+                                             node->job_name,
+                                             node->argc,
+                                             (const char **) node->argv);
+  if (job_data == NULL) {
+    /*
+      In this case the status of the job itself will be
+      unmodified; i.e. it will still be WAITING, and a new attempt
+      to submit it will be performed in the next round.
+    */
+    submit_status = SUBMIT_DRIVER_FAIL;
+    res_log_finfo("job data null on submit for %s(%d)",
```

Comment:
I guess a better log message would be something like: "Failed to submit job: %s(d)"?